### ISSUE TITLE
Support Lazarus

### ISSUE THREAD

**Issue by CFTechno** [amancini/ONVIF4Delphi#17]
Would supporting Lazarus be an option?

**Comment by amancini** [comment]
I'd like to add support for Lazarus and also for Android, but I've never had the chance to work on both, so I don't know much about them :(

**Comment by CFTechno** [comment]
yes, not that a straightforward 1-1 translation specially if you would like to support not just only Win OS but also Linux and MacOS

**Comment by amancini** [comment]
I'm trying to compile the project with Lazarus, but I saw that the XML parser, SHA1, BASE64 (which I think I adapted well for Lazarus) and HTTPClient need to be revised, there are quite a few changes :(

**Comment by CFTechno** [comment]
yes might need a few ifdefs , sha1 and base64 are available by default. httpclient needs to be fhttpclient. XML is also available  but needs a few units depending on which actions. https://wiki.lazarus.freepascal.org/XML_Tutorial
